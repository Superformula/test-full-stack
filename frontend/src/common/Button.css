.app-button {
    width: 280px;
    height: 90px;
    background: #FFFFFF;
    border: 4px solid rgba(0, 0, 0, 0.1);
    box-sizing: border-box;
    border-radius: 8px;
    font-style: normal;
    font-weight: 600;
    font-size: 24px;
    line-height: 30px;
    align-items: center;
    text-align: center;
    cursor: pointer;
    margin: auto;
}

.app-button:hover {
    border: 4px solid rgba(0, 0, 0, 0.4);
    box-sizing: border-box;
    border-radius: 8px;
    background: #F8F8F8;
}

.app-button-container {
    width: 100%;
    display: flex;
    margin: 50px 0;
}
