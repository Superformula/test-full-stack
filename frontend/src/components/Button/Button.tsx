import React from 'react';
import classnames from 'classnames';
import styles from './Button.module.scss';

type Props = {
  children?: React.ReactNode;
  variant?: 'primary' | 'secondary';
  onClick?: () => void;
};

const Button: React.FC<
  Props & React.ButtonHTMLAttributes<HTMLButtonElement>
> = ({ children, type = 'button', variant = 'primary', onClick }) => (
  <button
    type={type}
    className={classnames(styles.button, styles[`button--${variant}`])}
    onClick={onClick}
  >
    {children}
  </button>
);

export default Button;
