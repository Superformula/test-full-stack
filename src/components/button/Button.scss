@import "src/styles/variables";

.button {
  width: 280px;
  height: 90px;
  background: $color-white;
  color: $color-black;
  border: 4px solid rgba(0, 0, 0, 0.1);
  box-sizing: border-box;
  border-radius: $border-radius;

  font-weight: 600;
  font-size: 24px;
  line-height: 30px;
  display: flex;
  align-items: center;
  justify-content: center;

  text-transform: capitalize;
  cursor: pointer;
  &:hover,
  &:focus {
    border: 4px solid rgba(0, 0, 0, 0.4);
  }

  &:disabled {
    border: 4px solid rgba(0, 0, 0, 0.1);
    color: rgba(0, 0, 0, 0.5);
  }
}
