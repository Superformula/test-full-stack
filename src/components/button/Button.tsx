import { memo, MouseEventHandler } from 'react';
import { ChildrenProps } from '../General';
import classnames from 'classnames';

import classes from './Button.module.scss';

interface ButtonProps extends ChildrenProps {
  secondary?: boolean;
  disabled?: boolean;
  onClick?: MouseEventHandler;
  type?: 'submit' | 'reset' | 'button';
}

function ButtonComponent(props: ButtonProps) {
  return (
    <button
      type={props.type}
      className={classnames(classes.element, {
        [classes.secondary]: props.secondary,
      })}
      disabled={props.disabled}
      onClick={props.onClick}
    >
      {props.children}
    </button>
  );
}

export const Button = memo(ButtonComponent);
