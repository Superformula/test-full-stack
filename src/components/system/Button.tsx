import styled from "@emotion/styled";

type Props = {
  $variant?: "primary" | "secondary";
  $error?: boolean;
};

export const Button = styled.button<Props>`
  align-items: center;
  background-color: ${({ theme, $variant }) =>
    $variant === "secondary" ? theme.color.bgHaze : theme.color.bgPrimary};
  border-radius: 8px;
  border: 4px solid
    ${({ theme, $error }) => ($error ? theme.color.error : theme.color.divider)};
  color: ${({ theme, $error }) =>
    $error ? theme.color.error : theme.color.primary};
  cursor: pointer;
  display: flex;
  font-size: 1.5rem;
  font-weight: 600;
  height: 5.625rem;
  justify-content: center;
  line-height: 1.875rem;
  text-transform: uppercase;
  transition: border 0.3s ease-in-out;
  width: 17.5rem;

  :hover:not(:disabled) {
    border-color: ${({ theme }) => theme.color.hover};
  }
  :focus {
    outline: none;
    border-color: ${({ theme }) => theme.color.focus};
  }
  :disabled {
    cursor: not-allowed;
    color: ${({ theme }) => theme.color.disabled};
  }
`;
