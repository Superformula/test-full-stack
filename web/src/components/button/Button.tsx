import classNames from 'classnames'
import React, { ReactNode, useMemo } from 'react'
import style from './Button.module.scss'

declare type ButtonType = 'primary' | 'default'

declare type ButtonProps = React.HTMLProps<HTMLButtonElement> & {
  type?: ButtonType
  children: ReactNode
}

export const Button = ({ type, children, ...props }: ButtonProps) => {
  const classes = useMemo(
    () =>
      classNames({
        [style.button]: true,
        [style.default]: type === 'default' || !type,
        [style.primary]: type === 'primary' || !type,
      }),
    [type],
  )
  return (
    <button className={classes} {...props}>
      {children}
    </button>
  )
}
