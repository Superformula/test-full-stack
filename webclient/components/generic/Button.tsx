import { ReactElement, ButtonHTMLAttributes } from 'react'

import styles from './Button.module.css'

export default function Button({
  children,
  ...props
}: ButtonHTMLAttributes<HTMLButtonElement>): ReactElement {
  return (
    <div className={styles.wrapper}>
      <button {...props} className={styles.button}>
        {children}
      </button>
    </div>
  )
}
