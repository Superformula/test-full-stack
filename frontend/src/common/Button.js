import React from 'react'
import './Button.css'

export const Button = ({ label, onClick }) => {
  return (
    <div className={'app-button-container'}>
      <button className={'app-button'} onClick={onClick}>
        {label}
      </button>
    </div>
  )
}
