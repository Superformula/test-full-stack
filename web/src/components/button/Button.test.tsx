import { render } from '@testing-library/react'
import React from 'react'
import { Button } from './Button'

test('renders button', () => {
  const { container } = render(<Button type="primary">Hello World</Button>)
  expect(container).toMatchSnapshot()
})
