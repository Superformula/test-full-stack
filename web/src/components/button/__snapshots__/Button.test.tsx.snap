// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button 1`] = `
<div>
  <button
    class="button primary"
  >
    Hello World
  </button>
</div>
`;
